File: backend/schemas/maintenance.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class MaintenanceCreate(BaseModel):
    asset_id:       int
    assigned_to:    int
    type:           str
    priority:       str = 'medium'
    title:          str
    description:    str = ''
    scheduled_date: str
    notes:          Optional[str] = ''


class MaintenanceUpdate(BaseModel):
    status:       Optional[str]   = None
    completed_at: Optional[str]   = None
    cost:         Optional[float] = None
    notes:        Optional[str]   = None


class MaintenanceOut(MaintenanceCreate):
    id:            int
    status:        str
    asset_name:    Optional[str] = None
    assigned_name: Optional[str] = None
    completed_at:  Optional[datetime] = None
    cost:          Optional[float]    = None
    created_at:    datetime
    updated_at:    datetime
    model_config = {"from_attributes": True}

File: src/types/Maintenance.ts
export type MaintenanceStatus   = 'scheduled' | 'in_progress' | 'completed' | 'cancelled' | 'overdue'
export type MaintenancePriority = 'low' | 'medium' | 'high' | 'critical'
export type MaintenanceType     = 'preventive' | 'corrective' | 'inspection' | 'emergency'

export interface MaintenanceLog {
  id: number
  asset_id: number
  asset_name?: string
  assigned_to: number
  assigned_name?: string
  type: MaintenanceType
  priority: MaintenancePriority
  status: MaintenanceStatus
  title: string
  description: string
  scheduled_date: string
  completed_at: string | null
  cost: number | null
  notes: string
  created_at: string
  updated_at: string
}

export interface CreateMaintenancePayload {
  asset_id: number
  assigned_to: number
  type: MaintenanceType
  priority: MaintenancePriority
  title: string
  description: string
  scheduled_date: string
  notes?: string
}

export const STATUS_COLORS: Record<MaintenanceStatus, string> = {
  scheduled:   '#4f7cff',
  in_progress: '#f5a623',
  completed:   '#22c48a',
  cancelled:   '#bbb',
  overdue:     '#f05252',
}

export const PRIORITY_COLORS: Record<MaintenancePriority, string> = {
  low:      '#22c48a',
  medium:   '#f5a623',
  high:     '#f05252',
  critical: '#7c3aed',
}